### ISSUE TITLE
Floating point logging format gives false error

### ISSUE THREAD

**Issue by ebbek** [PyCQA/pylint#9118]
### Bug description

```python
Pylint issues the error logging-too-many-args for the following file (named logerr.py:

#! /usr/bin/env python3
"""
Demonstrate wrong message from pylint
"""
import logging
logging.warning( "The frequency is: %f MHz", 2.3 )
```

### Configuration

```ini
[LOGGING]
logging-format-style=new
```

### Command used

```shell
pylint logerr.py
```

### Pylint output

```python
************* Module logerr
logerr.py:8:0: E1205: Too many arguments for logging format string (logging-too-many-args)
```

### Expected behavior

No errors.

### Pylint version

```shell
pylint 3.0.0
astroid 3.0.0
Python 3.11.6 (tags/v3.11.6:8b6ee5b, Oct  2 2023, 14:57:12) [MSC v.1935 64 bit (AMD64)]
```

### OS / Environment

Windows 11

**Comment by Pierre-Sassoulas** [comment]
To reproduce you also need:
```ini
[LOGGING]
logging-format-style=new
```
Opened #9121
